packages/mwp-router/src/util/resolve.js

```diff
@@ -42,30 +42,39 @@ export const getChildRoutes = (
 export const resolveChildRoutes = (
 	matchedRoute: MatchedRoute
 ): Promise<Array<PlatformRoute>> => {
-	const { match } = matchedRoute;
+	const { match, route } = matchedRoute;
 	if (match.isExact) {
-		return _resolveIndexRoute(matchedRoute).then(
-			m => (m.route.indexRoute ? [m.route.indexRoute] : [])
+		return _resolveIndexRoute(route).then(
+			r => (r.indexRoute ? [r.indexRoute] : [])
 		);
 	}
-	return _resolveNestedRoutes(matchedRoute).then(m => m.route.routes || []);
+	return _resolveNestedRoutes(route).then(r => r.routes || []);
 };
 
-const _resolveNestedRoutes: Resolver<MatchedRoute> = matchedRoute =>
-	matchedRoute.route.getNestedRoutes
-		? matchedRoute.route
-				.getNestedRoutes()
-				.then(routes => (matchedRoute.route.routes = routes))
-				.then(() => matchedRoute)
-		: Promise.resolve(matchedRoute);
+const _resolveNestedRoutes: Resolver<PlatformRoute> = ({
```

Comment:
these functions were refactored to be 'pure' and deal only with routes, not 'matchedroutes' - they return a _new route object_ - the original route is not modified like it was previously